doc/Authoring_Extensions.md

```diff
@@ -0,0 +1,250 @@
+            execute: (): any => {
+                this.messageService.info('Hello World!');
+                return null;
+            },
+            isEnabled: () => true
+        });
+    }
+}
+...
+```
+
+Note how we use `@inject` in the constructor to get the `MessageService` as a property, and how we use that later in the implementation of the handler. This is the elegance of dependency injection: As a client, we neither care where these dependencies come from nor what their lifecycle is.
+
+To make it accessible by the UI, we implement a `MenuContribution`, adding an item to the Search/Replace section of the edit menu in the menu bar.
+
+```typescript
+...
+@injectable()
+export class HelloWorldMenuContribution implements MenuContribution {
+
+    registerMenus(menus: MenuModelRegistry): void {
+        menus.registerMenuAction([
+            MAIN_MENU_BAR,
+            CommonMenus.EDIT_MENU,
+            CommonMenus.EDIT_MENU_FIND_REPLACE_GROUP
+        ], {
+                commandId: HelloWorldCommand.id,
+                label: 'Say Hello'
+            });
+    }
+}
+```
+
+## Executing the Extension In the Browser
+
+Now we want to see our extension in action. For this purpose, the generator has created a `package.json` in the folder `browser-app`. It defines a Theia browser application with a couple of statically included extensions, including our `hello-world-extension`. All remaining files in this directory have been auto-generated by `yarn` calling the `theia-cli` tool during the build, as defined in the scripts section.
+
+```json
+{
+  "name": "browser-app",
+  "version": "0.1.0",
+  "dependencies": {
+    "@theia/core": "next",
+    "@theia/filesystem": "next",
+    "@theia/workspace": "next",
+    "@theia/preferences": "next",
+    "@theia/navigator": "next",
+    "@theia/process": "next",
+    "@theia/terminal": "next",
+    "@theia/editor": "next",
+    "@theia/languages": "next",
+    "@theia/markers": "next",
+    "@theia/monaco": "next",
+    "@theia/typescript": "next",
+    "hello-world-extension": "0.1.0"
+  },
+  "devDependencies": {
+    "@theia/cli": "next"
+  },
+  "scripts": {
+    "prepare": "theia build",    
+    "start": "theia start",
+    "watch": "theia build --watch"
+  },
+  "theia": {
+    "target": "browser"
+  }
+}
+```
+
+Now we have all pieces together to build and run the application. 
+To run the browser app, enter:
+
+```bash
+cd browser-app
+yarn start
+```
+
+and point your browser to http://localhost:3000. Then choose _Edit > Say Hello_ from the menu: A message "Hello World!" should pop up.
+
+## Executing the Extension In Electron
+
+The `package.json` for the Electron app looks almost the same, except for the name and the target property.
+
+```json
+{
+  "name": “electron-app",
+  ...
+  "theia": {
+    "target": "electron"
+  }
+}
+```
+
+Before running the electron app, you additionally have to rebuild some native modules:
+
+```bash
+yarn rebuild:electron
```

Comment:
@akosyakov is this already handled automatically, or not yet?

Reply:
not yet